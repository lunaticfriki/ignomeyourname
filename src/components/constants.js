// header
const headerTitle = 'i gnome your name!'
const headerSubtitle = 'Search your favorite gnome from Brastlewark!'

//nav
const navSearcher = 'Searcher'
const navAbout = 'About me'

//main
const mainTitle = ''
const mainubtitle = ''

// searcher
const sideSearcherTitle = 'SEARCHER'

//footer
const footerSignature = 'oscar ventura -'
const footerMessage = 'gnome searcher'

export {
  headerTitle,
  headerSubtitle,
  navSearcher,
  navAbout,
  mainTitle,
  mainubtitle,
  sideSearcherTitle,
  footerSignature,
  footerMessage
}
