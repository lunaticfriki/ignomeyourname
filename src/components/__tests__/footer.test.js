import React from 'react'
import { mount } from 'enzyme'
import renderer from 'react-test-renderer'

import Footer from '../footer'
import { translations } from '../constants'

describe('Footer component suite', () => {
  const { footerSignature, footerMessage, year } = translations
  const wrapperMount = mount(<Footer />)
  test('Footer snapshot', () => {
    const component = renderer.create(<Footer />)
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  test('Should render a Footer component', () => {
    expect(wrapperMount).toBeTruthy()
  })
  test('Should render the texts', () => {
    expect(wrapperMount.find('i').text()).toEqual(`${footerSignature} ${footerMessage} - ${year}`)
  })
})
