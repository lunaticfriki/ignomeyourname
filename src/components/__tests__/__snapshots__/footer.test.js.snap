// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component suite Footer snapshot 1`] = `
<div
  className="sc-AxjAm kRtxVN"
>
  <i>
    oscar ventura - gnome searcher - 2020
  </i>
</div>
`;
